### PR TITLE
[devicelab][windows] agent manager for windows.

### PR DESCRIPTION
**Pull request by godofredoc**
This is a script the will be running on a cron job every 5 minutes to
validate the agent is running on windows machines and restart it if
necessary. We will be implementing similar scripts for Linux and Mac
machines.

### CODE REVIEW

**Comment by godofredoc** [comment]
This PR fixes: 

https://github.com/flutter/flutter/issues/44209

File: devicelab/agent_manager.ps1 [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  Schtasks /create /tn "Agent Manager" /sc MINUTE /mo 5 /tr "PowerShell c:\Users\flutter\cocoon\devicelab\agent_manager.ps1"
+#>
+
+#---------------------------------------------------------[Initialisations]--------------------------------------------------------
+
+#----------------------------------------------------------[Declarations]----------------------------------------------------------
+$AgentFolder = "C:\Users\flutter\cocoon\agent"
+$LogFile = "C:\Users\flutter\agent_manager.log"
+$StdErr = "C:\Users\flutter\agent_stderr.log"
+$StdOut = "C:\Users\flutter\agent_stdout.log"
```
Comment (digiter) [inline devicelab/agent_manager.ps1]: Combining stdout and stderr into one log file makes it easier to check errors. Only one file needs to be checked.

File: devicelab/agent_manager.ps1 [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#-----------------------------------------------------------[Functions]------------------------------------------------------------
+function ReStart-Agent {
```
Comment (digiter) [inline devicelab/agent_manager.ps1]: Nit: Restart-Agent

File: devicelab/agent_manager.ps1 [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  $Process = Get-Process | Where-Object ProcessName -Match "^dart*"
+  if (!$process) {
+    echo "Restarting Agent"
+    cd $AgentFolder
+    Start-Process C:\tools\dart-sdk\bin\dart -ArgumentList "bin\agent.dart ci" -RedirectStandardOutput $StdOut -RedirectStandardError $StdErr -Verbose
```
Comment (digiter) [inline devicelab/agent_manager.ps1]: It would be nice to see when the process started in logs.

File: devicelab/agent_manager.ps1 [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  } else {
+    echo "Agent already running. Skipping."
```
Comment (digiter) [inline devicelab/agent_manager.ps1]: There will be lots of this in the logs, I think it's ok to remove it. With the above comment, the logs will only records when the manager tries to start an agent.

File: devicelab/agent_manager.ps1 [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#requires -version 2
```
Comment (digiter) [inline devicelab/agent_manager.ps1]: Is there a style guide? It's better to keep the style consistent.

Comment (digiter) [reply devicelab/agent_manager.ps1]: Looks unnecessarily verbose, ¯\_(ツ)_/¯

**Pull request merged**